framework/src/NormalisedSumPDF.cpp

```diff
@@ -293,121 +304,81 @@ bool NormalisedSumPDF::SetPhysicsParameters( ParameterSet * NewParameterSet )
 //Return the integral of the function over the given boundary
 double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBoundary * NewBoundary )
 {
-	(void)NewDataPoint;
-	(void)NewBoundary;
-	//The evaluate method already returns a normalised value
+	// Calculate and cache the integrals
+	double firstIntegralResult = firstPDF->Integral( NewDataPoint, NewBoundary ) * firstIntegralCorrection;
+	double secondIntegralResult = secondPDF->Integral( NewDataPoint, NewBoundary ) * secondIntegralCorrection;
+	SetCachedIntegrals({firstIntegralResult,secondIntegralResult}, NewDataPoint, NewBoundary);
```

Comment:
Technically if you ask a PDF to integrate over an observable it doesn't know about it can't. I suppose you could be asking to integrate over a sub-space of the global phase-space (i.e. component plotting) but you should tell the PDF first you're changing the phase-space and then integrate over it imo which is what ComponentPlotter _should_ be doing.

It would be nice to see if these PhaseSpaces aren't equivalent as this could help track down some difficult to understand bugs.